docs/aws.md

```diff
@@ -0,0 +1,150 @@
+# AWS Setup Instructions
+The goal of this document is to provide a guide on how to set up the GitHub Action runner on AWS. This document will guide you through the account setup required to use this action with AWS.
+
+## Prerequisites
+- An AWS account
+
+## Setup
+1. Prepare an IAM user with AWS access keys
+    1. Sign in to your AWS Management Console.
+    2. Go to the IAM console.
+    3. In the navigation pane, choose "Policies" and click "Create Policy".
+    4. Select the "JSON" tab, paste the following JSON, and click "Next":
+        ```json
+        {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Action": [
+                "ec2:RunInstances",
+                "ec2:TerminateInstances",
+                "ec2:DescribeInstances",
+                "ec2:DescribeInstanceStatus"
+              ],
+              "Resource": "*"
+            }
+          ]
+        }
+        ```
+    5. Name the policy (e.g., `gha-runner-policy`) and click "Create Policy".
+    6. In the navigation pane, choose "Users" and click "Add user".
+    7. Enter a username (e.g., `gha-runner-user`) and ensure "Provide user access to the AWS Management Console" is unchecked.
+    8. Select "Attach policies directly", search for the policy you created (`gha-runner-policy`), and select it.
+    9. Click "Create user".
+    10. Click on the newly created user, go to the "Security credentials" tab, and click "Create access key".
+    11. Choose "Other", click "Next", and then copy the Access Key and Secret Access Key (or download the CSV file). These keys will not be shown again.
+2. Create your GitHub Access Token
+    1. This can be done with either a Personal Access Token or a Fine-Grained Personal Access Token.
+    2. Go to your GitHub account settings.
+    3. Click on "Developer settings".
+    4. Create a new token with `repo` scope.
```

Comment:
I'm having a little trouble with this - Our repos are owned by our org, so making a personal access token, I don't have access to openforcefield org repos (in this case I only have access to `openforcefield/openff-evaluator` - my personal fork of its feedstock.)

<img width="928" alt="Screenshot 2024-07-17 at 5 18 44 PM" src="https://github.com/user-attachments/assets/bed9fe89-a26e-439d-a212-d8feb8d442cc">

Reply:
Oh, the org level settings have to be updated to allow create of PATs for the org:

<img width="1528" alt="Screenshot 2024-07-17 at 5 25 38 PM" src="https://github.com/user-attachments/assets/4b3440a3-edbb-4b05-a5e2-ae2957afe53f">

Reply:
You may also be able to set the resource owner to the org as well.

Reply:
Yeah, it looks like I can set the org as the resource owner now that I've allowed PATs at the org level. 

The next issue I hit is that the "repo" scope is an option on the "classic" token creator page, but that page doesn't let me select the org as the resource owner (I'm thinking that this would work, but the token would then cover BOTH the org and my personal repos). 

On the other hand, the "beta" token creator page does let me assign the org as resource owner, but it's not clear which permissions on this page are equivalent to "repo" from the classic token creator page. I'd love some guidance on that. 

I'm heading out for the night but will try to pick this back up tomorrow.

Reply:
I largely opt to use fine-grained Tokens instead of classic tokens as that is the direction that GitHub seems to be moving. I will update these docs to include the fine-grained access token permissions needed (I have what needs to be added just have to write it up).

Thanks for working on this!!

Reply:
One annoyance about fine-grained tokens is that they expire. I wrote a little actions workflow that will fail when your token should be replaced: https://github.com/omsf-eco-infra/scripts/blob/main/.github/workflows/check_token.yml